promql/engine.go

```diff
@@ -1524,12 +1525,23 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, annotations.Annotatio
 				if len(outVec) > 0 {
 					if outVec[0].H == nil {
 						if ss.Floats == nil {
-							ss.Floats = getFPointSlice(numSteps)
+							if prevSS != nil && cap(prevSS.Floats)-2*len(prevSS.Floats) > 0 {
```

Comment:
Suggest extract this section to a function whose name explains what is going on.
The same set of lines appears to be repeated later.

Reply:
PTAL?